### PR TITLE
New learn menu

### PR DESCRIPTION
**Pull request by pettinarip**
## Description

New Learn menu layout + some new styles.

- The `Dropdown` now doesn't define the layout of the items. It is only rendering the `children`.
- `Menu` is defining the layout of the items using the `Dropdown` component.

TODO:
- [nice to have] migrate to TS

Related to https://github.com/ethereum/ethereum-org-website/issues/6285

### CODE REVIEW

File: src/components/Nav/Dropdown.js [hunk 1]
```diff
@@ -140,54 +144,53 @@ const NavDropdown = ({ section, hasSubNav }) => {
...
-        ) : (
+    <DropdownContext.Provider
+      value={{ isOpen, toggle, close, tabInteractionHandler }}
+    >
+      <NavListItem ref={ref} aria-label={translateMessageId(ariaLabel, intl)}>
+        <DropdownTitle
+          dir={isPageRightToLeft ? "auto" : "ltr"}
+          onClick={() => toggle()}
+          onKeyDown={onKeyDownHandler}
+          tabIndex="0"
```
Comment (minimalsm) [inline src/components/Nav/Dropdown.js:155]: We don't currently signify that this is an interactive element.

```suggestion
          tabIndex="0"
          role="button"
          aria-expanded={isOpen ? "true" : "false"}
```

**Review by minimalsm — COMMENTED** [review]
This is looking great :-)!

File: src/components/Nav/Dropdown.js [hunk 2]
```diff
@@ -140,54 +144,53 @@ const NavDropdown = ({ section, hasSubNav }) => {
...
+        >
+          <Translation id={section.text} />
           <StyledIcon isOpen={isOpen} name="chevronDown" />
-        )}
-      </DropdownTitle>
+        </DropdownTitle>
 
-      {section.items ? (
         <DropdownList
```
[the remark this answers is not in the document]
Comment (pettinarip) [reply src/components/Nav/Dropdown.js]: Good point. I had to do it in the inner div. Do you think there is a problem with that? or should we do it in the ul for some reason?

File: src/components/Nav/Dropdown.js (hunk 1 above)
Comment (pettinarip) [reply src/components/Nav/Dropdown.js]: Nice! done

File: src/components/Nav/Dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+import Icon from "../Icon"
+import Link from "../Link"
+
+import { useOnClickOutside } from "../../hooks/useOnClickOutside"
+import { getDirection, translateMessageId } from "../../utils/translations"
+import { Lang } from "../../utils/languages"
+
+import { ISection } from "./types"
+
+// TODO use framer-motion
```
Comment (samajammin) [inline src/components/Nav/Dropdown.tsx:16]: Hah fun memories 😅

File: src/components/Nav/Dropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+const DropdownItem = styled.li`
+  margin: 0;
+  color: ${(props) => props.theme.colors.text};
+  &:hover {
+    color: ${(props) => props.theme.colors.primary};
+    background: ${(props) => props.theme.colors.dropdownBackgroundHover};
+  }
+`
+
+// TODO move to SharedStyles
```
Comment (samajammin) [inline src/components/Nav/Dropdown.tsx:86]: 😅

**Review by samajammin — APPROVED** [review]
Awesome! 🎉

**Review by wackerow — APPROVED** [review]

**Pull request merged**